core/newlib_syscalls.c

```diff
@@ -303,6 +303,10 @@ void _lock_acquire(_lock_t *lock) {
 
 void _lock_acquire_recursive(_lock_t *lock) {
     if (locks_initialized) {
+        if (sdk_NMIIrqIsOn) {
+            uart_putc(0, ':');
```

Comment:
Is this `uart_putc` call really necessary?

Reply:
Thought was that if it gets here it is an fatal error path, so emit some diagnostic info. A restart is going to be verbose anyway. Welcome to omit them.